src/app/features/courses/courses.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input } from '@angular/core';
+import { CourseDataType } from '@app/shared/types/course.model';
+
+@Component({
+  selector: 'app-courses',
+  templateUrl: './courses.component.html',
+  styleUrls: ['./courses.component.scss']
+})
+export class CoursesComponent {
+  @Input() courses:CourseDataType[]=[];
+  showCourseInfo:CourseDataType | undefined=undefined;
+
+  readonly emptyListTitle = 'Your List is Empty';
+  readonly emptyListText = "Please use 'ADD NEW COURSE' button to add your first course";
+
+  handleShowCourse(courseId:CourseDataType['id']) {
```

Comment:
Please mark the return type (a boolean) and once the Array.find returns a course, set it to true. Otherwise false.

Reply:
Ignore my first comment. Fn does not return a boolean value, but it is a void function. But we can mark this, of course.
Please read on with typescript fn signatures, especially the return types here: https://www.educba.com/typescript-function-return-type/

Reply:
oh i get it. It is inferred as void ,  but it may be clearer to define the void explicitly. ->changed